orchestrator/cosmos_gravity/src/utils.rs

```diff
@@ -56,3 +69,180 @@ impl BadSignatureEvidence {
         }
     }
 }
+
+#[allow(clippy::too_many_arguments)]
+/// Collects the needed Gravity.sol balances for EthereumClaim submissions by first learning which
+/// contracts to monitor, then collecting all the ethereum heights needed and then finally
+/// querying the ERC20 balances required, packing it all up into a Map
+pub async fn collect_eth_balances_for_claims(
```

Comment:
as discussed in previous comments this has to be grabbed individually for each claim to deal with the situation that the list has changed in the middle of playing back some attestations. A variant of the batch stress test where we change the list.